iocore/net/QUICNetVConnection.cc

```diff
@@ -1134,7 +1134,7 @@ QUICNetVConnection::_store_frame(ats_unique_buf &buf, size_t &len, bool &retrans
   frame->debug_msg(msg, sizeof(msg));
   QUICConDebug("[TX] %s", msg);
 
-  frame->store(buf.get() + len, &l);
+  frame->store(buf.get() + len, &l, max_size);
```

Comment:
We should check the returned value if we do this, but what should we do if it returns 0?

Reply:
We can build a new packet and send it out because the buffer doesn't have enough space for this frame.

Reply:
If there was no bugs, we should be able to store the frame always except one case that `maximum_quic_packet_size ` is shrunk by PMTUD which is not implemented yet.

Line 1117 checks that `frame` fits in the packet by calculating `len + frame->size() + MAX_PACKET_OVERHEAD`. If it does, we should be able to store the frame. If it doesn't, a new buffer for a new packet will be allocated on line 1127, and we probably be able to store the frame into the new buffer.

Since frame size is already fixed here, we may need to split the frame into two frames if we get 0 here.

Reply:
How about fixing size when generate stream frame (Also retransmitter should split retransmit stream frame)?  I mean giving a limit size to generate_frame, and FrameGenerator should not generate the frame which is larger than this size. Then store operation should be always succeed .

Reply:
>  I mean giving a limit size to generate_frame, and FrameGenerator should not generate the frame which is larger than this size.

We already have that mechanism, don't we? If not, I agree, we should do that.

> Then store operation should be always succeed .

This is not always true. As I wrote on the last comment, maximum packet size can be changed. If the size was changed after generating frames, the frame may not fit in new smaller packets.